tests/test_analytics_http_client.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+from unittest.mock import patch, MagicMock
+import httpx
+from usage_analytics import AnalyticsHttpClient, ANALYTICS_TO_TRACK
+
+
+class TestAnalyticsHttpClient(unittest.TestCase):
+    def setUp(self):
+        self.module_name = "test_module"
+        self.client = AnalyticsHttpClient(module_name=self.module_name)
+
+    def test_init(self):
+        """Test that the client initializes with the correct module name."""
+        self.assertEqual(self.client.module_name, self.module_name)
+        self.assertIsInstance(self.client, httpx.Client)
+
+    @patch("usage_analytics.analytics_http_client.super")
+    def test_request_adds_headers(self, mock_super):
+        """Test that request method adds analytics headers."""
+        mock_request = MagicMock()
+        mock_super.return_value.request = mock_request
+
+        # Call request method
+        self.client.request(
+            "GET", "https://example.com", headers={"Existing": "Header"}
```

Comment:
You can use this enum to avoid typing out the requests types. This enables autocompletion. And, although veeeery unlikely, that you get the correct GET verbs if the global community wants to change it to `GET2`.

https://docs.python.org/3/library/http.html#http.HTTPMethod

Reply:
Unfortunately this is only available from python 3.11